ops/charm.py

```diff
@@ -239,3 +269,15 @@ class PayloadMeta:
     def __init__(self, name, raw):
         self.payload_name = name
         self.type = raw['type']
+
+
+class FunctionMeta:
+    """An object containing metadata about an function"""
+
+    def __init__(self, name, raw=None):
+        raw = raw or {}
+        self.function_name = name
+        self.title = raw.get('title', '')
+        self.description = raw.get('description', '')
+        self.parameters = raw.get('params', {})  # <parameter name>: <JSON-schema>
```

Comment:
It actually is a data structure conforming to the [JSON Schema specification](https://json-schema.org/specification.html) ([per docs](https://discourse.jujucharms.com/t/actions-for-the-charm-author/1113#heading--options-and-format-actionsyaml)).

Maybe something like this would be more descriptive:

```python
        self.parameters = raw.get('params', {})  # Mapping of names to JSON Schema definitions
```